plugins/modules/keycloak_client.py

```diff
@@ -750,7 +750,7 @@ def sanitize_cr(clientrep):
     :param clientrep: the clientrep dict to be sanitized
     :return: sanitized clientrep dict
     """
-    result = clientrep.copy()
+    result = copy.deepcopy(clientrep)
     if 'secret' in result:
         result['secret'] = 'no_log'
```

Comment:
What exactly is the deepcopy needed for? You mentioned that this PR is because `secret` is replaced by `no_log` - but the shallow `clientrep.copy()` should already prevent the original dict's secret value to be modified. Is there something else that is modified (for example in `normalise_cr`)?

Reply:
It's the attribute ``saml.signing.private.key``. The ``clientrep.copy()`` keep the reference for the sub dict ``attributes``.

https://github.com/ansible-collections/community.general/blob/867aee606e42ad52e8ea653d890397b033c8bf7d/plugins/modules/keycloak_client.py#L758

Reply:
:+1: